src/System.Runtime.Loader/tests/ContextualReflection.cs

```diff
@@ -0,0 +1,1055 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.Loader;
+using System.Runtime.Remoting;
+using System.Threading.Tasks;
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+
+namespace System.Runtime.Loader.Tests
+{
+    class AGenericClass<T>
+    {
+    }
+
+    class MockAssembly : Assembly
+    {
+        public MockAssembly() {}
+    }
+
+    public class ContextualReflectionTestFixture : IContextualReflectionTestFixture
+    {
+        public AssemblyLoadContext isolatedAlc { get; }
+        public Assembly isolatedAlcAssembly { get; }
+        public Type isolatedAlcFixtureType { get; }
+        public IContextualReflectionTestFixture isolatedAlcFixtureInstance { get; }
+
+        public AssemblyLoadContext defaultAlc { get; }
+        public Assembly defaultAlcAssembly { get; }
+        public Type defaultAlcFixtureType { get; }
+
+        [MethodImplAttribute(MethodImplOptions.NoInlining)]
```

Comment:
It should not be necessary to mark these with NoInlining attribute. `CORINFO_FLG_DONT_INLINE_CALLER` flag in the JIT should ensure that these are not getting inlined.